### PR TITLE
rename dfs to tree search,and make a template for tree search

### PR DESCRIPTION
**Pull request by evykassirer**
😍

### CODE REVIEW

File: lib/TreeSearch.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// Returns a function that performs a postorder search on the tree for the given
+// simplifcation function
+TreeSearch.postOrder = function(simplificationFunction) {
+  return function (node) {
+    return search(simplificationFunction, node, false);
+  };
+};
+
+// A helper function for performing a tree search with a function
+function search(simplificationFunction, node, preOrder) {
```
Comment (aelnaiem) [inline lib/TreeSearch.js]: 😍

File: lib/collectAndCombine.js [hunk 2]
```diff
@@ -166,7 +133,7 @@ function multiplyLikeTerms(node) {
     return status;
   }
 
-  status = Fraction.multiplyFractionsDFS(node);
+  status = Fraction.multiplyFractionsTreeSearch(node);
```
Comment (aelnaiem) [inline lib/collectAndCombine.js]: Why not just call it multiplyFractions? All the other simplifications that are tree searches are just exported by the module so they are only referred to by their name e.g. when used collectAndCombineLikeTermsTreeSearch is actually just collectAndCombineLikeTerms

File: lib/print.js [hunk 3]
```diff
@@ -10,25 +10,25 @@ const NodeType = require('./NodeType');
...
-function printDFS(node) {
+function printTreeTraveral(node) {
   if (PolynomialTermNode.isPolynomialTerm(node)) {
     const polyTerm = new PolynomialTermNode(node);
     // This is so we don't print 2/3 x^2 as 2 / 3x^2
     // Still print x/2 as x/2 and not 1/2 x though
     if (polyTerm.hasFractionCoeff() && node.op !== '/') {
       const coeffTerm = polyTerm.getCoeffNode();
-      const coeffStr = printDFS(coeffTerm);
+      const coeffStr = printTreeTraveral(coeffTerm);
```
Comment (aelnaiem) [inline lib/print.js]: Traversal*

File: lib/collectAndCombine.js (hunk 2 above)
Comment (evykassirer) [inline lib/collectAndCombine.js]: I'll just change this in the next PR

**Pull request merged**